service/history/shard/context_impl.go

```diff
@@ -1176,7 +1176,9 @@ func (s *ContextImpl) transitionLocked(request contextRequest) {
 		s.state = contextStateStopping
 		// The change in state should cause all write methods to fail, but just in case, set this also,
 		// which will cause failures at the persistence level
-		s.shardInfo.RangeId = -1
+		if s.shardInfo != nil {
```

Comment:
An alternate approach would be to always initialize shardInfo with a dummy (non-nil) value, which could protect against other uses before it's initialized (I couldn't find any though). But I can change it to do it that way instead.

Reply:
Put a comment here explaining when this could happen.